src/app/WriteClient.h

```diff
@@ -120,12 +120,18 @@ class WriteClient : public Messaging::ExchangeDelegate
      *  @param[in]    apExchangeMgr    A pointer to the ExchangeManager object.
      *  @param[in]    apCallback       Callback set by application.
      *  @param[in]    aTimedWriteTimeoutMs If provided, do a timed write using this timeout.
+     *  @param[in]    aSuppressResponse If provided, set SuppressResponse field to the provided value
      */
-    WriteClient(Messaging::ExchangeManager * apExchangeMgr, Callback * apCallback,
-                const Optional<uint16_t> & aTimedWriteTimeoutMs) :
+    WriteClient(Messaging::ExchangeManager * apExchangeMgr, Callback * apCallback, const Optional<uint16_t> & aTimedWriteTimeoutMs,
+                const Optional<bool> & aSuppressResponse = NullOptional) :
```

Comment:
why optional for a bool? should it not be just true/false? There should be some default somewhere.

for type safety it is probably better to have some enum like `SuppressResponse::kYes` or `kSuppressResponse` than a bool.